### PR TITLE
Extract CardViewController form CardComponent

### PR DESCRIPTION
**Pull request by descorp**
### Changes

* extract CardViewController from CardComponent

### CODE REVIEW

File: AdyenCard/Components/Card/CardViewController.swift [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        button.showsActivityIndicator = true
+        view.isUserInteractionEnabled = false
+    }
+
+    public func update(binInfo: BinLookupResponse) {
+        self.securityCodeItem.update(cardBrands: binInfo.brands ?? [])
+
+        guard let brands = binInfo.brands else { return }
+        self.numberItem.showLogos(for: brands.isEmpty ? self.topCardTypes : brands.map(\.type))
+    }
```
Comment (mohammedDehairy) [inline AdyenCard/Components/Card/CardViewController.swift:103]: why are those public? the whole class is internal, right?

File: AdyenCard/Components/Card/CardComponentExtensions.swift [hunk 2]
```diff
@@ -143,6 +143,7 @@ extension CardComponent {
         ///   Defaults to true.
         ///   - storedCardConfiguration: Stored card configuration.
         ///   - supportedCardTypes: The supported card types.
+        ///   - showsAddressVerification: Indicates whether to show address verification section.  Defaults to false.
```
[the remark this answers is not in the document]
Comment (descorp) [reply AdyenCard/Components/Card/CardComponentExtensions.swift]: ups, leftover 😅

File: AdyenCard/Components/Card/CardViewController.swift (hunk 1 above)
Comment (descorp) [reply AdyenCard/Components/Card/CardViewController.swift:103]: Yep, that is a good point

File: AdyenCard/Components/Card/CardViewController.swift [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    private let configuration: CardComponent.Configuration
+    private let formStyle: FormComponentStyle
+    private let payment: Payment?
+    private let environment: Environment
+    private let supportedCardTypes: [CardType]
+    private let scope: String
+
+    private let maxCardsVisible = 4
+    private let publicBinLenght = 6
+    private let throttler = Throttler(minimumDelay: 0.5)
```
Comment (mohammedDehairy) [inline AdyenCard/Components/Card/CardViewController.swift:23]: I prefer one empty line between ivars, what do you think?

File: Tests/AdyenTests/Card Tests/BCMCComponentTests.swift [hunk 4]
```diff
@@ -144,7 +144,7 @@ class BCMCComponentTests: XCTestCase {
         self.populate(textItemView: cardNumberItemView!, with: "00000")
         DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + .seconds(1)) {
 
-            XCTAssertTrue(cardNumberItem.cardTypeLogos.allSatisfy { $0.isHidden })
+            XCTAssertTrue(cardNumberItem.cardTypeLogos.allSatisfy { !$0.isHidden })
```
Comment (mohammedDehairy) [inline Tests/AdyenTests/Card Tests/BCMCComponentTests.swift:147]: why is the condition flipped here?

**Review by mohammedDehairy — APPROVED** [review]

File: Tests/AdyenTests/Card Tests/BCMCComponentTests.swift (hunk 4 above)
Comment (descorp) [reply Tests/AdyenTests/Card Tests/BCMCComponentTests.swift]: Good point. Updated

**Review by mohammedDehairy — APPROVED** [review]

**Pull request merged**